mmv1/third_party/tgc_next/test/hcl_test.go

```diff
@@ -0,0 +1,172 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var (
+	basicHCL = `
+resource "google_project_service" "project" {
+  service = "iam.googleapis.com"
+}
+`
+	nestedBlocksHCL = `
+resource "google_storage_bucket" "bucket" {
+  name          = "my-bucket"
+  location      = "US"
+  force_destroy = true
+
+  lifecycle_rule {
+    action {
+      type = "Delete"
+    }
+    condition {
+      age = 30
+    }
+  }
+}
+`
+	multipleResourcesHCL = `
+resource "google_project_service" "project" {
+  service = "iam.googleapis.com"
+}
+
+resource "google_storage_bucket" "bucket" {
+  name = "my-bucket"
+}
+`
+	listOfNestedObjectsHCL = `
+resource "google_compute_firewall" "default" {
+  name    = "test-firewall"
+  network = google_compute_network.default.name
+
+  allow {
+    protocol = "icmp"
+  }
+
+  allow {
+    protocol = "tcp"
+    ports    = ["80", "8080", "1000-2000"]
+  }
+
+  source_tags = ["web"]
+}
+`
+	listOfMultiLevelNestedObjectsHCL = `
```

Comment:
This test case is the same with `listOfNestedObjectsHCL`.  Should it be a different test case?

Reply:
Ah, yes it should. Adding that now.